framework/src/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.java

```diff
@@ -339,106 +339,126 @@ public TransferFunction createFlowTransferFunction(CFAbstractAnalysis<Value, Sto
+        if (!foundOtherwise && this.isSupportedQualifier(unqualified)) {
             defs.addCheckedCodeDefault(unqualified, DefaultLocation.OTHERWISE);
+            foundOtherwise = true;
+        }
+
+        if(!foundOtherwise) {
+            ErrorReporter
+                    .errorAbort("GenericAnnotatedTypeFactory.createQualifierDefaults: "
+                                        + "@DefaultQualifierInHierarchy or @DefaultFor(DefaultLocation.OTHERWISE) not found. "
+                                        + "Every checker must specify a default qualifier.");
         }
+    }
 
-        // Add defaults for unannotated code if conservative unannotated flag is passed and
-        // no defaults were given.
-        if ((// !checker.hasOption("unsafeDefaultsForUnannotatedBytecode")
-                // temporarily use unsafe defaults for bytecode, unless option given
-                checker.hasOption("safeDefaultsForUnannotatedBytecode") ||
-                // This block may need to be split after safeDefaults... is reverted to unsafeDefaults...
-                checker.hasOption("useSafeDefaultsForUnannotatedSourceCode")) &&
-                !foundDefaultOtherwiseForUnannotatedCode) {
-            Set<? extends AnnotationMirror> tops = this.qualHierarchy.getTopAnnotations();
-            for (AnnotationMirror top : tops) {
-                defs.addUncheckedCodeDefault(top, DefaultLocation.RETURNS);
-                defs.addUncheckedCodeDefault(top, DefaultLocation.UPPER_BOUNDS);
+    /**
+     * Adds default qualifiers for code that is not type-checked by
+     * reading  {@code @DefaultForUncheckedCode} and {@code @DefaultQualifierForUncheckedCode}
+     * meta-annotations. Then it applies the standard
+     * unchecked code defaults, if a default was not specified for a particular location.
+     * <p>
+     * Standard unchecked code default are: <br>
+     * top: {@code DefaultLocation.RETURNS,DefaultLocation.FIELD,DefaultLocation.UPPER_BOUNDS}<br>
+     * bottom: {@code DefaultLocation.PARAMETERS, DefaultLocation.LOWER_BOUNDS}<br>
+     * <p>
+     * If {@code @DefaultQualiferForUnchecked} code is not found or a default for {@code DefaultLocation.Otherwise}
+     * is not used, the defaults for checked code will be applied to locations without a default for unchecked code.
+     * <p>
+     * Subclasses may override this method to add defaults that cannot be specified with
+     * a {@code @DefaultForUncheckedCode} or {@code @DefaultQualifierForUncheckedCode} meta-annotations or to change
+     * the standard defaults.
+     *
+     * @param defs QualifierDefault object to which defaults are added
+     */
+    protected void addUncheckedCodeDefaults(QualifierDefaults defs) {
+        for (Class<? extends Annotation> annotation : getSupportedTypeQualifiers()) {
+            DefaultInUncheckedCodeFor defaultForUnannotated = annotation.getAnnotation(DefaultInUncheckedCodeFor.class);
+
+            if (defaultForUnannotated != null) {
```

Comment:
Recommend doing a scrub for all instances of "Unannotated" since we will soon forget what we meant by that term.